lib/api/apiUtils/quotas/quotaUtils.js

```diff
@@ -64,11 +65,124 @@ function isMetricStale(metric, resourceType, resourceName, action, inflight, log
             action,
             inflight,
         });
+        monitoring.requestWithQuotaMetricsUnavailable.inc();
         return true;
     }
     return false;
 }
 
+/**
+ * Evaluates quotas for a bucket and an account.
+ *
+ * @param {number} bucketQuota - The quota limit for the bucket.
+ * @param {number} accountQuota - The quota limit for the account.
+ * @param {object} bucket - The bucket object.
+ * @param {object} account - The account object.
+ * @param {number} inflight - The number of inflight requests.
+ * @param {number} inflightForCheck - The number of inflight requests for checking quotas.
+ * @param {string} action - The action being performed.
+ * @param {boolean} cleanUp - Flag indicating whether to ignore quotas.
```

Comment:
i don't think this flag is even necessary: we could just rely on the fact that inflight will be `< 0` in that case, and just ignore the quota check result: it will be the same number of request anyway...